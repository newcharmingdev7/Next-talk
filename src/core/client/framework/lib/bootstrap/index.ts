export {
  TalkContext,
  TalkContextConsumer,
  TalkContextProvider,
} from "./TalkContext";
export { default as createManaged } from "./createManaged";
export { default as withContext } from "./withContext";
