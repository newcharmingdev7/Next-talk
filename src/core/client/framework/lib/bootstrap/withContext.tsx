import React from "react";
import { hoistStatics, wrapDisplayName } from "recompose";

import { Omit } from "talk-ui/types";
import { TalkContext, TalkContextConsumer } from "./TalkContext";

// Injects props and removes them from the prop requirements.
// Will not pass through the injected props if they are passed in during
// render. Also adds new prop requirements from TNeedsProps.
type InferableComponentEnhancerWithProps<TInjectedProps> = <
  P extends TInjectedProps
>(
  component: React.ComponentType<P>
) => React.ComponentType<Omit<P, keyof TInjectedProps>>;

/**
 * withContext is a HOC wrapper around `TalkContextConsumer`.
 * `propsCallback` must be provided which accepts the `TalkContext`
 * and returns the props the should be injected.
 */
function withContext<T>(
  propsCallback: (context: TalkContext) => T
): InferableComponentEnhancerWithProps<T> {
  return hoistStatics<T>(
    <U extends T>(WrappedComponent: React.ComponentType<U>) => {
      const Component: React.StatelessComponent<any> = props => (
        <TalkContextConsumer>
          {context => (
            <WrappedComponent {...props} {...propsCallback(context)} />
          )}
        </TalkContextConsumer>
      );
      Component.displayName = wrapDisplayName(WrappedComponent, "withContext");
      return Component;
    }
  ) as any;
}

export default withContext;
