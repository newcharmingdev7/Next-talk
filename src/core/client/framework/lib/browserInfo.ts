import bowser from "bowser";

export interface BrowserInfo {
  ios: boolean;
}

export function getBrowserInfo(): BrowserInfo {
  return {
    ios: bowser.ios,
  };
}
