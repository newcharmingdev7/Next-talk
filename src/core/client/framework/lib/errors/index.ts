export { default as UnknownServerError } from "./unknownServerError";
export { default as InvalidRequestError } from "./invalidRequestError";
