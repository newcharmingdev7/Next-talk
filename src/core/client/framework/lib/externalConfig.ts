import { areWeInIframe } from "coral-framework/utils";
import { Child as PymChild } from "pym.js";

export interface ExternalConfig {
  accessToken?: string;
}

export function getExternalConfig(
  pym?: PymChild
): Promise<ExternalConfig> | null {
  if (pym && areWeInIframe()) {
    return new Promise(resolve => {
      pym.sendMessage("getConfig", "");
      pym.onMessage("config", raw => {
        resolve(JSON.parse(raw) as ExternalConfig);
      });
    });
  }
  return null;
}
