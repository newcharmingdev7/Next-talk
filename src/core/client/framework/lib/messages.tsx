import { Localized } from "fluent-react/compat";
import React from "react";

/**
 * This file contains localization messages that are shared by
 * different parts of the framework.
 */

export const VALIDATION_REQUIRED = () => (
  <Localized id="framework-validation-required">
    <span>This field is required.</span>
  </Localized>
);

export const VALIDATION_TOO_SHORT = () => (
  <Localized id="framework-validation-tooShort">
    <span>This field is too short.</span>
  </Localized>
);

export const INVALID_EMAIL = () => (
  <Localized id="framework-validation-invalidEmail">
    <span>Please enter a valid email address.</span>
  </Localized>
);

export const INVALID_USERNAME = () => (
  <Localized id="framework-validation-invalidUsername">
    <span>Please enter a valid username.</span>
  </Localized>
);

export const INVALID_PASSWORD = () => (
  <Localized id="framework-validation-invalidUsername">
    <span>Please enter a valid password.</span>
  </Localized>
);

export const PASSWORDS_DO_NOT_MATCH = () => (
  <Localized id="framework-validation-passwordsDoNotMatch">
    <span>Passwords do not match. Try again.</span>
  </Localized>
);
