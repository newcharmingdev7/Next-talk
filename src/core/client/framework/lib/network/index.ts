export { default as createNetwork, TokenGetter } from "./createNetwork";
export { default as extractError } from "./extractError";
