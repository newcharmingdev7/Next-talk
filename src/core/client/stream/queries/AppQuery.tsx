import * as React from "react";
import { StatelessComponent } from "react";
import { ReadyState } from "react-relay";

import {
  graphql,
  QueryRenderer,
  withLocalStateContainer,
} from "talk-framework/lib/relay";
import {
  AppQueryResponse,
  AppQueryVariables,
} from "talk-stream/__generated__/AppQuery.graphql";
import { AppQueryLocal as Local } from "talk-stream/__generated__/AppQueryLocal.graphql";

import PermalinkViewContainer from "talk-stream/containers/PermalinkViewContainer";
import AppContainer from "../containers/AppContainer";

interface InnerProps {
  local: Local;
}

interface WrappedProps {
  data: any;
}

// TODO (bc) Break down the needs of each component into another file (maybe).
// (careful porting QueryRenderer into another stateless component)

export const renderWrapper = (
  WrappedComponent: React.ComponentType<WrappedProps>
) => ({ error, props }: ReadyState<AppQueryResponse>) => {
  if (error) {
    return <div>{error.message}</div>;
  }
  if (props) {
    return <WrappedComponent data={props} />;
  }
  return <div>Loading</div>;
};

const AppQuery: StatelessComponent<InnerProps> = ({
  local: { commentID, assetID },
}) => {
  if (commentID) {
    return (
      <QueryRenderer<AppQueryVariables, AppQueryResponse>
        query={graphql`
          query AppQuery_PermalinkViewContainer_Query($commentID: ID!) {
            ...PermalinkViewContainerQuery @arguments(commentID: $commentID)
          }
        `}
        variables={{
          commentID,
        }}
        render={renderWrapper(PermalinkViewContainer)}
      />
    );
  }

  return (
    <QueryRenderer<AppQueryVariables, AppQueryResponse>
      query={graphql`
        query AppQuery($assetID: ID!) {
          ...AppContainer @arguments(assetID: $assetID)
        }
      `}
      variables={{
        assetID,
      }}
      render={renderWrapper(AppContainer)}
    />
  );
};

const enhanced = withLocalStateContainer<Local>(
  graphql`
    fragment AppQueryLocal on Local {
      assetID
      commentID
      assetURL
    }
  `
)(AppQuery);

export default enhanced;
