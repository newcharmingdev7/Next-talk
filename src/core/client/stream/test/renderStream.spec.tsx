import React from "react";
import TestRenderer from "react-test-renderer";
import { RecordProxy } from "relay-runtime";

import { timeout } from "talk-common/utils";
import { TalkContext, TalkContextProvider } from "talk-framework/lib/bootstrap";
import AppQuery from "talk-stream/queries/AppQuery";

import createEnvironment from "./createEnvironment";
import { assets } from "./fixtures";

const resolvers = {
  Query: {
    asset: () => assets[0],
  },
};

const environment = createEnvironment({
  // Set this to true, to see graphql responses.
  logNetwork: false,
  resolvers,
  initLocalState: (localRecord: RecordProxy) => {
    localRecord.setValue(assets[0].id, "assetID");
  },
});

const context: TalkContext = {
  relayEnvironment: environment,
  localeMessages: [],
};

const testRenderer = TestRenderer.create(
  <TalkContextProvider value={context}>
    <AppQuery />
  </TalkContextProvider>
);

it("renders comment stream", async () => {
  // Wait for loading.
  await timeout();
  expect(testRenderer.toJSON()).toMatchSnapshot();
});
