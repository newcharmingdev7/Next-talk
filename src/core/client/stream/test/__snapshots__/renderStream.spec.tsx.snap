// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders comment stream 1`] = `
<div
  className="root"
>
  <div>
    <h1
      className="root heading1 gutterBottom"
    >
      Talk NEO
    </h1>
    <form
      autoComplete="off"
      onSubmit={[Function]}
    >
      <div>
        <textarea
          className="textarea"
          name="body"
          onChange={[Function]}
          value=""
        />
      </div>
      <div
        className="postButtonContainer"
      >
        <button
          className="root root primary"
          disabled={false}
          onBlur={[Function]}
          onFocus={[Function]}
          onMouseDown={[Function]}
        >
          Post
        </button>
      </div>
    </form>
    <div
      className="root gutterBottom"
    >
      <p
        className="root body1 gutterBottom author"
      >
        Markus
      </p>
      <p
        className="root body1"
      >
        Joining Too
      </p>
    </div>
    <div
      className="root gutterBottom"
    >
      <p
        className="root body1 gutterBottom author"
      >
        Lukas
      </p>
      <p
        className="root body1"
      >
        What's up?
      </p>
    </div>
  </div>
</div>
`;
